client/reader/controller.js

```diff
@@ -58,14 +58,11 @@ function trackScrollPage( path, title, category, readerView, pageNum ) {
 
 // Listen for route changes and remove the full post dialog when we navigate away from it
 pageNotifier( function removeFullPostOnLeave( newContext, oldContext ) {
-	if ( ! oldContext ) {
-		return;
-	}
-
 	const fullPostViewRegex = /^\/read\/(blogs|feeds)\/([0-9]+)\/posts\/([0-9]+)$/i;
 
 	if ( ( ! oldContext || oldContext.path.match( fullPostViewRegex ) ) && ! newContext.path.match( fullPostViewRegex ) ) {
 		newContext.store.dispatch( hideReaderFullPost() );
+		removeFullPostDialog();
```

Comment:
I'm adding this line because we need to remove the full post dialog on navigating away from it. When there was no "Stream Not Found" page, it was possible to navigate away from the dialog only through closing it, which called the `removeFullPostDialog` method. But now, user can navigate to other parts of Calypso through (e.g.) sidebar, so we must make sure the full post dialog is always removed, no matter how it is closed.

Also, the comment
> // Listen for route changes and remove the full post dialog when we navigate away from it
has it written in it. :laughing: